### DISCUSSION TITLE
How to know Coral is working? Frigate running as container in K3s

### DISCUSSION THREAD

**Discussion by mattgmoser** [blakeblackshear/frigate#2210] [Ask A Question]
Hi all. I'm curious if there's an easy way to determine if Coral is actually being utilized. I have a 3-node Raspberry Pi cluster and am running Frigate as a container. My coral device is plugged into my 03 node, and I have affinities set up to ensure Frigate only runs on that node.

My configuration for Coral looks like:
		detectors:
		  coral:
		    type: edgetpu
		    device: usb

This did not work initially, until I enabled privileged mode on the deployment/container. Now it fires up and runs. I'm still not sure if it's actually using Coral though!

**Comment by ozett** [comment]
check on the frigate debug page the inference speed for the detector,
small value == more coral.

check the frigate logs. there it would be stated if a coral is found

**Reply by mattgmoser** [reply]
Thanks! I'm hoping this is a good sign:

root@kuberpi01:~# kubectl logs pods/frigate-6cd9b49cd6-tn6jc -n frigate | grep coral
[2021-11-09 16:32:52] detector.coral                 INFO    : Starting detection process: 213

![image](https://user-images.githubusercontent.com/46821505/141016687-20957cc5-7923-415c-93c8-77f5fd57f53e.png)

**Reply by blakeblackshear** [reply]
If you have specified a Coral in your config, frigate won't even run if it doesn't detect it.

**Reply by mattgmoser** [reply]
> If you have specified a Coral in your config, frigate won't even run if it doesn't detect it.

Awesome, that's great to hear. It's always another layer of fun involving containers. Thank you!

**Reply by ozett** [reply]
maybe i would be helpful for pci-TPUs to see the working temperature (beside the inference-time) as a heartbeat
https://coral.ai/docs/pcie-parameters/#pcie-parameters-overview
